diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SmartCastRule.kt

```diff
@@ -234,12 +234,12 @@ class SmartCastRule(private val configRules: List<RulesConfig>) : Rule("smart-ca
 
         val identifier = node.getFirstChildWithType(REFERENCE_EXPRESSION)?.text
 
-        node.getAllChildrenWithType(WHEN_ENTRY).forEach {
-            if (it.hasChildOfType(WHEN_CONDITION_IS_PATTERN) && identifier != null) {
-                val type = it.getFirstChildWithType(WHEN_CONDITION_IS_PATTERN)!!
+        node.getAllChildrenWithType(WHEN_ENTRY).forEach { node ->
```

Comment:
It's not a very good practice to shadow variable name from the outer scope. More appropriate name would be something like `entry`, since we are iterating through when entries.